velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -48,6 +48,20 @@
 #include <aws/s3/model/HeadObjectRequest.h>
 #include <aws/s3/model/UploadPartRequest.h>
 
+namespace {
+bool shouldRetryRequest(
```

Comment:
This can be a private function of S3ReadFile.

Reply:
I want to keep it outside so we may use it for other calls too.